src/charm.py

```diff
@@ -338,10 +387,9 @@ def stop_mongos_service(self):
     def restart_charm_services(self):
         """Restart mongos service."""
         container = self.unit.get_container(Config.CONTAINER_NAME)
-        container.stop(Config.SERVICE_NAME)
-
         container.add_layer(Config.CONTAINER_NAME, self._mongos_layer, combine=True)
         container.replan()
+        container.restart(Config.SERVICE_NAME)
```

Comment:
**note:** This will cause a double restart, is that what you want?

If the new layer is not the same as the original, `pebble replan` --> `pebble stop; pebble start`. Then you do `pebble restart` again. It's probably not the end of the world, but you might want to just have `container.restart` here.

Reply:
you're goat ty